src/shared/ui/picker.ts

```diff
@@ -145,3 +149,201 @@ export function verifySinglePickerOutput<T extends vscode.QuickPickItem>(choices
 
     return choices[0]
 }
+
+// TODO: Cache these results? Should we have a separate store? Can we also use a store with values from the explorer tree?
+// If we move this to a cache, we should remove the awsCallLogic and instead listen/send events to/from the cache.
+// (this should also be retooled to not use the request/response objects directly)
+export class IteratingAWSCallPicker<TRequest, TResponse> {
```

Comment:
I'm not sure if there's a good way to test this class (since it's a wrapper for pagination logic and the quick picker). Thoughts?